Added safe templating to instructor_analytics.html.

dsjen opened the pull request.

dsjen commented:
@dan-f @andy-armstrong

dan-f commented:
Looks good, but any idea what the value of `section_data['insights_message']` is? I assume it's probably a plain string, but it's possible that it's got HTML tags in it (like `<em>` or something similar).

dsjen commented:
Oh, it can have html in it (https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor/views/instructor_dashboard.py#L640).  I'll add an `HTML()` wrapper!

dan-f commented:
@dsjen could you push the interpolation into the HTML file?

dsjen commented:
Yes!  I like that idea.

dsjen commented:
Updated so the construction is done in the template now.

dan-f commented:
Looks like you have a failure related to this change.

robrap commented on lms/templates/instructor/instructor_dashboard_2/instructor_analytics.html:
```diff
@@ -1,6 +1,26 @@
+from openedx.core.djangolib.markup import HTML, Text
+%>
+
+<%page expression_filter="h"/>
```
Put this at the top of the file please.

robrap commented on lms/templates/instructor/instructor_dashboard_2/instructor_analytics.html:
```diff
@@ -1,6 +1,26 @@
+                        course_id=Text(section_data['course_id'])
+                    ))
+                ),
+                analytics_dashboard_name=Text(settings.ANALYTICS_DASHBOARD_NAME),
```
This `Text()` call is not needed, but shouldn't hurt.  I'd remove it though.

dsjen commented:
@benpatterson -- the coverage oddly decreased for this build.  I moved code python a file to an underscore template and the result is tested in instructor dashboard tests.  I'm not sure what to do!

benpatterson commented:
Hmm @dsjen honestly that shouldn't result in a build failure. Coveralls is configured not to fail on decreases:

![image](https://cloud.githubusercontent.com/assets/2004678/14023818/a1335658-f1bc-11e5-94e8-f764cd392b89.png)

Any idea if other folks' pull requests are failing for coverage decreases?

dsjen commented:
@benpatterson -- I poked around other PRs and don't see coveralls failing for them.  I'll try re-running to see if perhaps it was a glitch.

dsjen commented:
jenkins run python

dsjen commented:
Reviewers, this has been updated to address review comments.  Please review!

The drop in coverage is because I removed lines (which were covered and are now deleted rather).  I checked in with TestEng and the advice is to ignore the coveralls "failure" in this case.

dan-f commented:
:

andy-armstrong commented:
:+1:
